test/helpers/mock_server.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'webrick'
+
+module MockServer
+  extend self
+
+  def start(port)
+    pid = Thread.new do
+      TestServer.new(hostname: '0.0.0.0', port: port).start
+    end
+    sleep(1)
```

Comment:
Sleep statements are gross. Is there another way to check for server health?

Reply:
There isn't a helper function to check server health in WEBrick, so had to put back the old `poll_until_ready`.